go/types/sequence_cursor.go

```diff
@@ -32,16 +49,67 @@ func (cur *sequenceCursor) getItem(idx int) sequenceItem {
 	return cur.seq.getItem(idx)
 }
 
+// sync loads the sequence that the cursor index points to.
+// It's called whenever the cursor advances/retreats to a different chunk.
 func (cur *sequenceCursor) sync() {
 	d.PanicIfFalse(cur.parent != nil)
 	cur.seq = cur.parent.getChildSequence()
+	if cur.readAheadMap != nil {
+		// if this cursor uses read-ahead, reset the cache for the next chunk
+		cur.readAheadMap = make(map[int]chan sequence)
+		cur.readAheadNext = 0
+	}
+}
+
+var readAheadParallelism = runtime.NumCPU() * 8
+
+func minInt(x, y int) int {
+	return int(math.Min(float64(x), float64(y)))
```

Comment:
Can we add a util/math package? This function gets written every month or so. Also it seems like it would be better to save a cast and do the min/max check yourself instead of going through math.min/max.